boa_engine/src/vm/completion_record.rs

```diff
@@ -0,0 +1,35 @@
+//! An implementation of a `CompletionRecord` for Boa's VM.
+
+use crate::{JsError, JsResult, JsValue};
+
+/// An implementation of the ECMAScript's `CompletionRecord` [specification] for
+/// Boa's VM output Completion and Result.
+///
+/// [specification]: https://tc39.es/ecma262/#sec-completion-record-specification-type
+#[derive(Debug, Clone)]
+pub(crate) enum CompletionRecord {
```

Comment:
I think this should also be used in the `AsyncGenerator` logic.
More specifically, in `AsyncGeneratorRequest`, which has a `completion` field.

Maybe it could also be used in `Generator`, but I think we would need to unify the `generator_resume` and `generator_resume_abrupt` methods into one.

Reply:
There are a few places that I think we should be able to use `CompletionRecord`. The primary place being generators, but I hadn't initially made the change on this draft as there may be more changes that are needed to generator execution to adapt for the shift, which seemed like it would be really expanding the scope of these changes.

Reply:
Completely understandable. We can fix that in another PR